src/cdk/tree/control/flat-tree-control.ts

```diff
@@ -0,0 +1,47 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {BaseTreeControl} from './base-tree-control';
+
+/** Flat tree control. Able to expand/collapse a subtree recursively for FlatNode type. */
+export class FlatTreeControl<T> extends BaseTreeControl<T> {
+
+  /** Construct with flat tree data node functions getLevel and isExpandable. */
+  constructor(public getLevel: (dataNode: T) => number,
+              public isExpandable: (dataNode: T) => boolean) {
+    super();
+  }
+
+  /**
+   * Gets a list of descendent data nodes of a subtree rooted at given data node recursively.
```

Comment:
"Gets a list of the data node's subtree of descendent data nodes" (seems like its a given that its recursive?). What does it mean that the data nodes must be "set correctly"?

Reply:
Must be flattened tree nodes